dtcw

```diff
@@ -95,300 +152,476 @@ Examples:
-            doJavaCheck=false
-            echo "use docker installation"
+        # TODO: What is this good for? Has probably to do with Docker image creation.
+        if [[ "${DTC_HEADLESS}" = false ]]; then
+            # important for the docker file to find dependencies
+            DTC_OPTS="${DTC_OPTS} '-Dgradle.user.home=${DTC_ROOT}/.gradle'"
+        fi
+        if [ "${env}" = local ]; then
+            cmd="${DTC_HOME}/bin/doctoolchain . ${*} ${DTC_OPTS}"
         else
-
-            echo "docToolchain not installed."
-            if [ "${sdkman}" = true ]; then
-                echo "please use sdkman to install docToolchain"
-                echo "$ sdk install doctoolchain ${DTC_VERSION}"
-                exit
-            else
-                echo "sdkman not found"
-                if [ "${DTC_HEADLESS}" == true ]; then
-                    # just download
-                    install_local
-                else
-                    echo "Do you wish to install doctoolchain to '${HOME}/.doctoolchain'?"
-                    select yn in "Yes" "No"; do
-                        case ${yn} in
-                        Yes)
-                            echo "installing doctoolchain"
-                            install_local
-                            break
-                            ;;
-                        No)
-                            echo "you need docToolchain as CLI-Tool installed or docker."
-                            echo "to install docToolchain as CLI-Tool, please install"
-                            echo "sdkman and re-run this command."
-                            echo "https://sdkman.io/install"
-                            echo "\$ curl -s \"https://get.sdkman.io\" | bash"
-                            exit 1
-                            ;;
-                        esac
-                    done
-                fi
-                command="${HOME}/.doctoolchain/docToolchain-${DTC_VERSION}/bin/doctoolchain . ${*} ${DTC_OPTS}"
-            fi
+            cmd="$(sdk_home_doctoolchain "${version}")/bin/doctoolchain . ${*} ${DTC_OPTS}"
         fi
     fi
-fi
-if [ "${doJavaCheck}" = true ]; then
-    check_java
-fi
-
-# use x86 emulator on Apple M1/M2
-emu=""
-if [ "$os" = "Darwin" ] && [ "$arch" = "arm64" ]; then
-  echo "Apple silicon detected, run in x86 mode"
-  emu="arch -x86_64"
-fi
-
-# echo "Command to invoke: ${command}"
-exec ${emu} bash -c "$command"
+    echo "${cmd}"
+}
+
+show_os_related_info() {
+
+    # check if we are running on WSL
+    if grep -qsi 'microsoft' /proc/version &> /dev/null ; then
+        # TODO: bug - This URL leads to now-where!
+        printf "%s\n" \
+            "" \
+            "Bash is running on WSL which might cause problems with plantUML." \
+            "See https://doctoolchain.github.io/docToolchain/#wsl for more details." \
+            ""
```

Comment:
@rdmueller This link leads to the docToolchain landing page (is not valid anymore). I could not find information about WSL with plantuml anymore.

Reply:
correct link is https://doctoolchain.org/docToolchain/v2.0.x/010_manual/50_Frequently_asked_Questions.html#wsl
I'll fix this

Reply:
fixed